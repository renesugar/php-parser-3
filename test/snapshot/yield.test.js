const parser = require('../main');

describe("yield", function() {
  it("simple", function() {
    expect(parser.parseEval('yield $i;')).toMatchSnapshot();
  });
  it("assign", function() {
    expect(parser.parseEval('$data = yield $value;')).toMatchSnapshot();
  });
  it("assign (parens)", function() {
    expect(parser.parseEval('$data = (yield $value);')).toMatchSnapshot();
  });
  it("simple (key and value)", function() {
    expect(parser.parseEval('yield $id => $fields;')).toMatchSnapshot();
  });
  it("assign (key and value)", function() {
    expect(parser.parseEval('$data = (yield $key => $value);')).toMatchSnapshot();
  });
  it("inside function", function() {
    expect(parser.parseEval('function fn() { yield $i; }')).toMatchSnapshot();
  });
  it("null", function() {
    expect(parser.parseEval('yield;')).toMatchSnapshot();
  });
});
