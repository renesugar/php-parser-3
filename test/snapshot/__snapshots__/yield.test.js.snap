// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yield assign (key and value) 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": Variable {
          "byref": false,
          "curly": false,
          "kind": "variable",
          "name": "data",
        },
        "operator": "=",
        "right": Yield {
          "key": Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "key",
          },
          "kind": "yield",
          "parenthesizedExpression": true,
          "value": Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "value",
          },
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`yield assign (parens) 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": Variable {
          "byref": false,
          "curly": false,
          "kind": "variable",
          "name": "data",
        },
        "operator": "=",
        "right": Yield {
          "key": null,
          "kind": "yield",
          "parenthesizedExpression": true,
          "value": Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "value",
          },
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`yield assign 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": Variable {
          "byref": false,
          "curly": false,
          "kind": "variable",
          "name": "data",
        },
        "operator": "=",
        "right": Yield {
          "key": null,
          "kind": "yield",
          "value": Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "value",
          },
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`yield inside function 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [],
      "body": Block {
        "children": Array [
          ExpressionStatement {
            "expression": Yield {
              "key": null,
              "kind": "yield",
              "value": Variable {
                "byref": false,
                "curly": false,
                "kind": "variable",
                "name": "i",
              },
            },
            "kind": "expressionstatement",
          },
        ],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "fn",
      },
      "nullable": false,
      "type": null,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`yield null 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Yield {
        "key": null,
        "kind": "yield",
        "value": null,
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`yield simple (key and value) 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Yield {
        "key": Variable {
          "byref": false,
          "curly": false,
          "kind": "variable",
          "name": "id",
        },
        "kind": "yield",
        "value": Variable {
          "byref": false,
          "curly": false,
          "kind": "variable",
          "name": "fields",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`yield simple 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Yield {
        "key": null,
        "kind": "yield",
        "value": Variable {
          "byref": false,
          "curly": false,
          "kind": "variable",
          "name": "i",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
